matminer/featurizers/bandstructure.py

```diff
@@ -108,15 +109,28 @@ class BandFeaturizer(BaseFeaturizer):
     Featurizes a pymatgen band structure object.
     """
 
-    def __init__(self):
-        pass
+    def __init__(self, kpoints=None, atol=0.01, include_symeqks=False,
+                 weight=1):
+        self.atol = atol
+        self.include_symeqks = include_symeqks
+        self.kpoints = kpoints
+        self.weight = weight
 
     def featurize(self, bs):
         """
         Args:
             bs (pymatgen BandStructure or BandStructureSymmLine or their dict):
                 The band structure to featurize. To obtain all features, bs
                 should include the structure attribute.
+            kpoints ([1x3 numpy array]): list of fractional coordinates of
+                k-points at which energy is extracted.
+            atol (float): the absolute tolerance for fractional coordinate
+                below which two coordinates are considered the same
+            include_symeqks (bool): if True, when looking for energy of the
+                bands at given kpoints, symmetrically equivalent kpoints are
+                also considered (recommended to keep False for speed)
+            weight (int >= 1): if larger than 1, some features are multiplied
```

Comment:
remove this parameter; the goal of the featurizer should be to produce physically relevant features, not to balance the weight of various features. If someone is designing a distance metric or needs to normalize data a certain way, they should do it *after* running the featurizer. There are many ways to weight / normalize data and it is not the job of featurizers to do that.

Note that I will at some point start a "distance_metrics" package. There, if you want to define some default weights for judging band structure distance, you can do so.

Reply:
ok I will remove it and add it later to distance_metrics